src/compile/unit.ts

```diff
@@ -166,7 +166,8 @@ export class UnitModel extends Model {
           (channel === X && vlEncoding.has(encoding, X2)) ||
           (channel === Y && vlEncoding.has(encoding, Y2))
         ) {
-        const scale = _scale[channel] = initScale(channel, encoding[channel], config, mark,
+        const scale = _scale[channel] = initScale(
+          channel, encoding[channel] as ChannelDefWithScale, config, mark,
```

Comment:
Since this is the trade-off we have to make when we don't enumerate the channels, I'll accept this.